ethcore/src/client/test_client.rs

```diff
@@ -351,17 +370,116 @@ impl TestBlockChainClient {
 	pub fn set_history(&self, h: Option<u64>) {
 		*self.history.write() = h;
 	}
+
+    /// Create a signed transaction destined for `address`
+    /// `data` is an ethabi encoded function call
+    pub fn contract_call_tx(&self, id: BlockId, address: Address, data: Bytes) -> SignedTransaction {
+		let from = Address::default();
+		Transaction {
+			nonce: self.nonce(&from, id).unwrap_or_else(|| self.spec.engine.account_start_nonce(0)),
+			action: Action::Call(address),
+			gas: U256::from(50_000_000),
+			gas_price: U256::default(),
+			value: U256::default(),
+			data: data,
+		}.fake_sign(from)
+    }
+
+    /// Create a signed transaction to deploy a contract encoded in the `data` parameter 
+    pub fn contract_create_tx(&self, id: BlockId, address: Address, data: Bytes) -> SignedTransaction {
+		Transaction {
+			nonce: self.nonce(&address, id).unwrap_or_else(|| self.spec.engine.account_start_nonce(0)),
+			action: Action::Create,
+			gas: U256::from(50_000_000),
+			gas_price: U256::default(),
+			value: U256::default(),
+			data: data,
+		}.fake_sign(address)
+    }
+
+    /// Evaluate a transaction on a local EVM Interpreter
+	fn do_virtual_call(&self, env_info: &EnvInfo, state: &mut State<StateDB>, t: &SignedTransaction, analytics: CallAnalytics) -> Result<Executed, CallError> {
+		fn call<V, T>(
```

Comment:
Seems like a code duplication from reall `Client`, we should definitely refactor that. Please add a TODO if there is no way to re-use the code currently.